hazelcast-jet-core/src/test/java/com/hazelcast/jet/rocksdb/RocksMapTest.java

```diff
@@ -0,0 +1,165 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.rocksdb;
+
+import com.hazelcast.internal.serialization.InternalSerializationService;
+import com.hazelcast.jet.JetException;
+import com.hazelcast.jet.core.JetTestSupport;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import static java.util.Collections.emptyMap;
+
+public class RocksMapTest extends JetTestSupport {
+    private RocksDBStateBackend rocksDBStateBackend;
+    private InternalSerializationService serializationService;
+
+    @Before
+    public void init() {
+        serializationService = getJetService(createJetMember())
+                .createSerializationService(emptyMap());
+        rocksDBStateBackend = new RocksDBStateBackend(serializationService);
+    }
+
+    @After
+    public void cleanup() {
+        rocksDBStateBackend.deleteKeyValueStore();
+        serializationService.dispose();
+    }
+
+    @Test
+    public void test_getMap() {
+        RocksMap<String, Integer> rocksMap = rocksDBStateBackend.getMap();
+        Assert.assertNotNull("failed to create RocksMap ", rocksMap);
+    }
+
+    @Test
+    public void test_put_and_get() {
+        //Given
+        RocksMap<String, Integer> rocksMap = rocksDBStateBackend.getMap();
+        //When
+        rocksMap.put("key1", 1);
+        //Then
+        Assert.assertNotNull("the insert wasn't applied", rocksMap.get("key1"));
+        if (rocksMap.get("key1") != 1) {
+            Assert.fail("get doesn't return the correct value for key1");
+        }
+    }
+
+    //Assert.assertEquals doesn't accept nullable parameters
+    //It can't be used to check on the contents of a map
+    @Test
+    public void test_update() {
+        //Given
+        RocksMap<String, Integer> rocksMap = rocksDBStateBackend.getMap();
+        //When
+        rocksMap.put("key1", 1);
+        rocksMap.put("key1", 2);
+        //Then
+        Assert.assertNotNull("the insert wasn't applied", rocksMap.get("key1"));
+        if (rocksMap.get("key1") != 2) {
+            Assert.fail("get returns the old value for key1");
+        }
+    }
+
+    @Test
+    public void test_putAll() {
+        //Given
+        RocksMap<String, Integer> rocksMap = rocksDBStateBackend.getMap();
+        Map<String, Integer> map = new HashMap<>();
+        //When
+        map.put("key1", 1);
```

Comment:
the local map should be populated in the `Given` phase, it is not the thing under test.